lib/dice/src/lib.rs

```diff
@@ -39,22 +40,26 @@ mod trust_quorum_dhe_cert_tmpl;
 pub use crate::handoff::{AliasData, CertData, RngData, SpMeasureData};
 
 pub const SEED_LENGTH: usize = SECRETKEY_SEED_LENGTH;
-const REG_ADDR_NONSEC: u32 = 0x40000900;
-
-fn get_cdi_reg_slice() -> &'static mut [u32] {
-    // SAFETY: Dereferencing this raw pointer is necessary to read the CDI
-    // from the LPC55 DICE registers. This pointer will always reference a
-    // valid memory region as this is where the LPC55 maps the DICE
-    // registers. Support from the lpc55-pac would move the unsafe code
-    // into the pac library:
-    // https://github.com/lpc55/lpc55-pac/issues/15
-    // https://github.com/lpc55/lpc55-pac/pull/16
-    unsafe {
-        slice::from_raw_parts_mut(
-            REG_ADDR_NONSEC as *mut u32,
-            SEED_LENGTH / mem::size_of::<u32>(),
-        )
-    }
+
+/// Retrieves the bank of CDI registers from the SYSCON as a slice of volatile
+/// cells.
+fn get_cdi_reg_slice(
```

Comment:
Out of curiosity, why keep this as a separate function versus moving it into the `CdiReg` constructor?

Reply:
*shrug* I could go either way, I think I was mostly trying not to introduce unnecessary churn here when fixing the aliasing.